website/static/js/components/autocomplete.js

```diff
@@ -0,0 +1,171 @@
+     * Abstract methods - subclasses must implement *
+     ************************************************/
+
+    /** Populate `items` from the `data` returned in `fetchData` */
+    processData: function(data) {
+        throw new MethodNotDefined('processData');
+    },
+    /**
+     * Given a list of typeahead items, return a function that accepts a query
+     *  and a callback, and calls the callback with a list of matched items.
+     */
+    substringMatcher: function(items) {
+        throw new MethodNotDefined('substringMatcher');
+    },
+    /** Given a typeahead item, return HTML to be displayed in the dropdown. */
+    suggestionTemplate: function() {
+        throw new MethodNotDefined('suggestionTemplate');
+    },
+    /** Called when the form is submitted */
+    onSubmit: function() {
+        throw new MethodNotDefined('onSubmit');
+    },
+    /** Get the data at the provided URL, return a promise */
+    fetchData: function () {
+        return $.get(this.dataUrl).fail(
+            function() {
+                console.log('failed to retrieve data');
+            });
+    },
+    /** Initialize the Typeahead widget for the component */
+    initTypeahead: function () {
+        this.inputElement.typeahead({
+            hint: false,
+            highlight: true,
+            minLength: 0
+        }, {
+            // name: 'projectSearch' + nodeType + namespace,
+            displayKey: function(data) {
+                return data.value.node.title;
+            },
+            templates: {
+                suggestion: this.suggestionTemplate
+            },
+            source: this.substringMatcher(this.items())
+        });
+
+        this.inputElement.bind('typeahead:selected', function(event, item) {
+            // Call the parent viewModel's onSelected
+            this.onSelected(item.value);
+        }.bind(this));
+    },
+    /** Clear the selection */
+    onClear: function () {
+        this.inputElement.typeahead('val', '');
+        this.itemSelected(null);
+    },
+    /** Set the item as selected */
+    onSelected: function (item) {
+        this.itemSelected(item);
+    }
+});
+
+
+var draftRegistrationsSearchViewModel = function (params) {
+    baseSearchViewModel.apply(this, arguments);
+};
+$.extend(draftRegistrationsSearchViewModel.prototype, baseSearchViewModel.prototype, {
+    processData: function (data) {
+        this.items(data.draftRegistrations);
+    },
+    onSubmit: function () {
+        // Redirect the user to continue the draft registration
+        window.location = this.itemSelected().url;
+    },
+    suggestionTemplate: function(item) {
+        var dateUpdated = new $osf.FormattableDate(item.value.dateUpdated);
+        var dateCreated = new $osf.FormattableDate(item.value.dateCreated);
+        return '<p>' + item.value.node.title + '</p>' +
+            '<p><small class="m-l-md text-muted">' + 'Initiated by: ' + item.value.initiator.name + '</small></p>' +
```

Comment:
In general, we very strongly discourage generating raw HTML in knockout- all display logic/ markup should be handled by the view layer. (in a mako file directly, or using knockout templates similarly to how `log_templates` are handled)

In particular, as written, this trusts the values passed in and renders them to the template as raw markup. If you absolutely must write markup in knockout, use $osf.htmlEscape` on that user-entered values (title, initiator.name, etc).